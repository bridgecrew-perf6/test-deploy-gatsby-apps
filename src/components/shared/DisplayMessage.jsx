import React from "react"
import { Grid, Typography } from "@material-ui/core"
import { Link } from "gatsby"

export const DisplayMessage = ({ message, linkCaption, linkTo }) => {
  return (
    <Grid container justify="center" direction="column" alignItems="center">
      <Grid
        item
        xs={12}
        md={6}
        style={{
          border: "0.5px solid gray",
          padding: "40px",
          marginBottom: "10vh",
          marginTop: "10vh",
        }}
      >
        <Typography gutterBottom align="center" variant="subtitle1">
          {message}
        </Typography>
        <Grid item>
          <Link to={linkTo} style={{ color: "black" }}>
            <Typography align="center" variant="subtitle2">
              {linkCaption}
            </Typography>
          </Link>
        </Grid>
      </Grid>
    </Grid>
  )
}
