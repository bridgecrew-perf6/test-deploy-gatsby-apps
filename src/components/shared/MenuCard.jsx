import React from "react"
import {
  Card,
  Grid,
  IconButton,
  makeStyles,
  Typography,
} from "@material-ui/core"
import addButton from "../../images/add-button.svg"

export const MenuCard = ({ item, onAddItem }) => {
  const classes = useStyles()
  return (
    <Card style={{ padding: "10px" }}>
      <Grid container justify="space-between" spacing={5} direction="column">
        <Grid item>
          <Typography variant={"subtitle2"}>{item.name}</Typography>
        </Grid>
        <Grid item>
          <Grid container justify="space-between" alignItems="center">
            <Grid item>
              <Typography variant="body2">{item.price}</Typography>
            </Grid>
            <Grid item>
              <IconButton
                component="span"
                size="small"
                className={classes.iconButton}
                onClick={() => onAddItem(item)}
              >
                <img
                  src={addButton}
                  alt={"add-item"}
                  style={{ width: "6vw", height: "6vh" }}
                />
              </IconButton>
            </Grid>
          </Grid>
        </Grid>
      </Grid>
    </Card>
  )
}

const useStyles = makeStyles(theme => ({
  iconButton: {
    "&:hover": { backgroundColor: "transparent" },
  },
}))
