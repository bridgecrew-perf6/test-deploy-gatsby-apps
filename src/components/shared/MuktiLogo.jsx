import React from "react"
import { navigate } from "gatsby"
import logo from "../../images/logo.svg"
import { IconButton, makeStyles } from "@material-ui/core"

export const MuktiLogo = () => {
  const classes = useStyles()
  return (
    <IconButton
      disableFocusRipple
      disableRipple
      className={classes.iconButton}
      onClick={() => navigate("/")}
    >
      <img src={logo} className={classes.logo} alt={"logo"} />
    </IconButton>
  )
}
const useStyles = makeStyles(theme => ({
  iconButton: {
    "&:hover": { backgroundColor: "transparent" },
  },
  logo: {
    height: "12vh",
    width: "12vw",
  },
}))
