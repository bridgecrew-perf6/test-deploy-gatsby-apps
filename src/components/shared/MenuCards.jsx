import React from "react"
import { Grid, Typography } from "@material-ui/core"
import { MenuCard } from "./MenuCard"

export const MenuCards = ({ heading, text, items }) => {
  return (
    <div style={{ padding: "40px" }}>
      <Grid container direction="column" justify="center" alignItems="center">
        <Typography variant="h3">{heading}</Typography>
        <Typography variant="subtitle1" align="center">
          {text}
        </Typography>
      </Grid>
      <Grid container spacing={9} style={{ marginTop: "10vh" }}>
        {items.map((eat, key) => {
          return (
            <Grid key={key} item xs={12} md={4}>
              <MenuCard name={eat.name} price={eat.price} />
            </Grid>
          )
        })}
      </Grid>
    </div>
  )
}
