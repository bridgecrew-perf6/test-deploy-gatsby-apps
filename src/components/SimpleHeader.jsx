import React from "react"
import { AppBar, Grid } from "@material-ui/core"
import { MuktiLogo } from "./shared/MuktiLogo"

export const SimpleHeader = () => {
  return (
    <div data-cy={"without-menu"} style={{ flexGrow: "1" }}>
      <AppBar position="fixed" style={{ boxShadow: "0px 0px 0px 0px" }}>
        <Grid container justify="center">
          <MuktiLogo />
        </Grid>
      </AppBar>
    </div>
  )
}
