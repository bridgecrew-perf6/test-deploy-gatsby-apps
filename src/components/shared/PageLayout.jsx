import React from "react"
import { HeaderWithMenu } from "../HeaderWithMenu"
import { Grid, makeStyles, MuiThemeProvider } from "@material-ui/core"
import { theme } from "../../theme"
import { AppFooter } from "../Footer"
import { SimpleHeader } from "../SimpleHeader"

const routes = [
  { name: "DRINKS", path: "/drinks" },
  { name: "EATS", path: "/eats" },
  { name: "ABOUT", path: "/about" },
]

export const PageLayout = ({ children, hideMenu }) => {
  console.log(hideMenu)
  const classes = useStyles()
  return (
    <MuiThemeProvider theme={theme}>
      <Grid container justify={"center"}>
        <Grid item xs={12}>
          {!hideMenu && <HeaderWithMenu routes={routes} />}
          {hideMenu && <SimpleHeader />}
        </Grid>
        <Grid item xs={12} style={{ paddingTop: "20vh" }}>
          <Grid container justify={"center"}>
            <Grid item xs={12}>
              <main>{children}</main>
            </Grid>
          </Grid>
        </Grid>
        <Grid item xs={12} className={classes.footerContainer}>
          <AppFooter />
        </Grid>
      </Grid>
    </MuiThemeProvider>
  )
}

const useStyles = makeStyles(theme => ({
  footerContainer: {
    marginTop: theme.spacing(5),
  },
  footer: {
    paddingTop: theme.spacing(3),
    paddingBottom: theme.spacing(3),
  },
}))
