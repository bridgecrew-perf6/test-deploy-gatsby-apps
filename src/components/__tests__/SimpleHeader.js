import React from "react"
import renderer from "react-test-renderer"
import { SimpleHeader } from "../SimpleHeader"
import { render, screen } from "@testing-library/react"
import { MuktiLogo } from "../shared/MuktiLogo"

describe("SimpleHeader component Snapshot", () => {
  it("renders SimpleHeader correctly", () => {
    const tree = renderer.create(<SimpleHeader />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  test("Render Mukti logo button on header ", () => {
    render(<MuktiLogo />)
    expect(screen.getByRole("button")).toBeInTheDocument()
  })
})
