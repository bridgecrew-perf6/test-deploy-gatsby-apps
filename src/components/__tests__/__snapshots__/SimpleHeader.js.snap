// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleHeader component Snapshot renders SimpleHeader correctly 1`] = `
<div
  style={
    Object {
      "flexGrow": "1",
    }
  }
>
  <header
    className="MuiPaper-root MuiAppBar-root MuiAppBar-positionFixed MuiAppBar-colorPrimary mui-fixed MuiPaper-elevation4"
    style={
      Object {
        "boxShadow": "0px 0px 0px 0px",
      }
    }
  >
    <div
      className="MuiGrid-root MuiGrid-container MuiGrid-justify-xs-center"
    >
      <button
        className="MuiButtonBase-root MuiIconButton-root makeStyles-iconButton-1"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="button"
      >
        <span
          className="MuiIconButton-label"
        >
          <img
            alt="logo"
            className="makeStyles-logo-2"
            src="test-file-stub"
          />
        </span>
      </button>
    </div>
  </header>
</div>
`;
